vendor/sigs.k8s.io/cluster-api/pkg/controller/machine/controller.go

```
@@ -128,11 +142,39 @@ func (c *MachineControllerImpl) Reconcile(machine *clusterv1.Machine) error {
 	}
 	if exist {
 		glog.Infof("reconciling machine object %v triggers idempotent update.", name)
-		return c.update(m)
+		err = c.update(m)
+		if err != nil {
+			requeueErr, ok := err.(*RequeueAfterError)
+			if ok {
+				glog.Infof("actuator returned requeue after error: %v", requeueErr)
+				c.enqueueAfter(machine, requeueErr.RequeueAfter)
+				return nil
+			}
+		}
+		return err
 	}
 	// Machine resource created. Machine does not yet exist.
 	glog.Infof("reconciling machine object %v triggers idempotent create.", m.ObjectMeta.Name)
-	return c.create(m)
+	err = c.create(m)
+	if err != nil {
+		requeueErr, ok := err.(*RequeueAfterError)
+		if ok {
+			glog.Infof("actuator returned requeue after error: %v", requeueErr)
+			c.enqueueAfter(machine, requeueErr.RequeueAfter)
```

Same thing here about returning rather than swallowing the error.